pydatalab/src/pydatalab/routes/v0_1/export.py

```diff
@@ -0,0 +1,161 @@
+import os
+import uuid
+from datetime import datetime, timezone
+from pathlib import Path
+
+from flask import Blueprint, current_app, jsonify, send_file
+from flask_login import current_user
+
+from pydatalab.config import CONFIG
+from pydatalab.export_utils import create_eln_file
+from pydatalab.models.export_task import ExportStatus, ExportTask
+from pydatalab.mongo import flask_mongo
+from pydatalab.permissions import active_users_or_get_only
+from pydatalab.scheduler import export_scheduler
+
+EXPORT = Blueprint("export", __name__)
+
+
+@EXPORT.before_request
+@active_users_or_get_only
+def _(): ...
+
+
+def _do_export(task_id: str, collection_id: str):
+    try:
+        flask_mongo.db.export_tasks.update_one(
+            {"task_id": task_id}, {"$set": {"status": ExportStatus.PROCESSING}}
+        )
+
+        export_dir = Path(CONFIG.FILE_DIRECTORY) / "exports"
+        export_dir.mkdir(exist_ok=True)
+
+        output_path = export_dir / f"{task_id}.eln"
+        create_eln_file(collection_id, str(output_path))
+
+        flask_mongo.db.export_tasks.update_one(
+            {"task_id": task_id},
+            {
+                "$set": {
+                    "status": ExportStatus.READY,
+                    "file_path": str(output_path),
+                    "completed_at": datetime.now(tz=timezone.utc),
+                }
+            },
+        )
+
+    except Exception as e:
+        flask_mongo.db.export_tasks.update_one(
+            {"task_id": task_id},
+            {
+                "$set": {
+                    "status": ExportStatus.ERROR,
+                    "error_message": str(e),
+                    "completed_at": datetime.now(tz=timezone.utc),
+                }
+            },
+        )
+
+
+def _generate_export_in_background(task_id: str, collection_id: str, app):
+    if app is not None:
+        with app.app_context():
+            _do_export(task_id, collection_id)
+    else:
+        _do_export(task_id, collection_id)
+
+
+@EXPORT.route("/collections/<string:collection_id>/export", methods=["POST"])
+def start_collection_export(collection_id: str):
+    from pydatalab.permissions import get_default_permissions
+
+    collection_exists = flask_mongo.db.collections.find_one({"collection_id": collection_id})
+    if not collection_exists:
+        return jsonify({"status": "error", "message": "Collection not found"}), 404
+
+    collection_with_perms = flask_mongo.db.collections.find_one(
+        {"collection_id": collection_id, **get_default_permissions(user_only=True)}
+    )
+    if not collection_with_perms:
+        return jsonify({"status": "error", "message": "Access denied"}), 403
```

Comment:
```suggestion
    collection_with_perms = flask_mongo.db.collections.find_one(
        {"collection_id": collection_id, **get_default_permissions(user_only=True)}
    )
    if not collection_with_perms:
        return jsonify({"status": "error", "message": "Collection not found"}), 404
```

Better not to leak IDs that exist but the user does not have permission for, so just treat missing ID as 404

Reply:
Can probably also set `user_only=False`